Userland/Libraries/LibCore/System.cpp

```diff
@@ -1748,6 +1764,18 @@ ErrorOr<String> current_executable_path()
     // Ignore error if it wasn't a symlink
     if (ret == -1 && errno != EINVAL)
         return Error::from_syscall("readlink"sv, -errno);
+#elif defined(AK_OS_GNU_HURD)
+    // We could read /proc/self/exe, but why rely on procfs being mounted
+    // if we can do the same thing procfs does and ask the proc server directly?
+    process_t proc = getproc();
+    if (!MACH_PORT_VALID(proc))
+        return Error::from_syscall("getproc"sv, -errno);
+    kern_return_t err = proc_get_exe(proc, getpid(), path);
```

Comment:
This is still using our 4KiB stack buffer, which is a secret PATH_MAX usage. Is there a PATH_MAX mitigation for proc_get_exe as well? :)

Reply:
Very very very unfortunately, the "doesn't limit path lengths" thing is just a public facade; in practice paths are currently transmitted over IPC as the `string_t` type, which is (drumroll)

```c
typedef char string_t [1024];
```
There are literally comments saying `/* XXX string_t LOSES! */` ("loses" apparently being GNU for "sucks") scattered over glibc source code; I'm not exaggerating.

There is hope for improvement in this area; [with some recent GNU MIG work, we might get real dynamically sized strings](https://mail.gnu.org/archive/html/bug-hurd/2023-04/msg00300.html) instead of this atrocity.

But yeah, currently `proc_get_exe()` just returns a `string_t` 😭